shtdlib.sh

```diff
@@ -1914,6 +1914,42 @@ function uri_unparser {
     echo "${working_uri}"
 }
 
+## Uniform Resource Identifier (URI) Hostname to Fully Qualified Domain Name (FQDN)
+# Opportunistically resolves the hostname portion of a URI, and replaces it
+# with a FQDN using the Name Service Switch (nsswitch) library hosts database.
+# If URI hostname resolves, or if no match is found, then it uses the unresolved
+# hostname of the original URI. Returns status code 1 if URI fails to parse.
+## Example
+# $ uri_hostname_to_fqdn http://app:8080
+# http://app.example.com:8080
+function uri_hostname_to_fqdn {
+    uri="${*}"
+    uri_parser ${uri} || return 1
+
+    local fqdns=$(getent hosts ${uri_host})
+
+    # If hostname exists in hosts library, return
+    if $(echo ${fqdns} | egrep -q "(^| )${uri_host}( |$)"); then
```

Comment:
I don't think you need the subshell, 

echo ${fqdns} | egrep -q "(^| )${uri_host}( |$)" ; then

should be sufficient